src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -782,17 +749,51 @@ private void WriteDynamicComplexProperty(object propertyValue, IEdmTypeReference
             serializer.WriteObjectInline(propertyValue, edmType, writer, nestedWriteContext);
         }
 
-        private void WriteExpandedNavigationProperties(
-           SelectExpandNode selectExpandNode,
-            ResourceContext resourceContext,
-            ODataWriter writer)
+        private void WriteComplexProperties(SelectExpandNode selectExpandNode, ResourceContext resourceContext, ODataWriter writer)
         {
+            Contract.Assert(selectExpandNode != null);
             Contract.Assert(resourceContext != null);
             Contract.Assert(writer != null);
 
-            IDictionary<IEdmNavigationProperty, ExpandedNavigationSelectItem> navigationPropertiesToExpand = selectExpandNode.ExpandedProperties;
+            var complexProperties = selectExpandNode.SelectedComplexes;
```

Comment:
>var [](start = 12, length = 3)

Prefer not to use var; can we use actual type (here, and in other places in this class?)  I realize this is how the old code worked, but it would be nice to clean up uses of "var" when we encounter them.